examples/common/chip-app-server/Server.cpp

```diff
@@ -159,22 +159,62 @@ void InitServer(AppDelegate * delegate)
         SuccessOrExit(err = gRendezvousServer.Init(params, &gTransports));
     }
 
-    // TODO: advertise this only when really operational once we support both
-    // operational and commisioning advertising is supported.
+    if (ConfigurationMgr().IsFullyProvisioned())
     {
-        constexpr uint64_t kTestFabricId = 5544332211;
-        err                              = Mdns::ServiceAdvertiser::Instance().Advertise(Mdns::OperationalAdvertisingParameters()
-                                                                .SetFabricId(kTestFabricId)
+        uint64_t fabricId;
+
+        if (ConfigurationMgr().GetFabricId(fabricId) != CHIP_NO_ERROR)
+        {
+            ChipLogError(Discovery, "Fabric ID not known. Using a default");
+            fabricId = 5544332211;
+        }
+
+        err = Mdns::ServiceAdvertiser::Instance().Advertise(Mdns::OperationalAdvertisingParameters()
+                                                                .SetFabricId(fabricId)
                                                                 .SetNodeId(chip::kTestDeviceNodeId)
                                                                 .SetPort(CHIP_PORT)
-#if INET_CONFIG_ENABLE_IPV4
+                                                                .EnableIpV4(true));
```

Comment:
Why is `EnableIpV4` becoming unconditional?

Reply:
Testing only really ... if you enable ipv4 we advertise that we have it.
We could make it optional or have additional config variables.

Also harsh reality for me: in my wifi setup, somehow ipv4 is a ton more reliable than ipv6, so until we make our ipv6 solid and testable, having dual v4 and v6 is probably easier on developers.